components/site/src/link_checking.rs

```diff
@@ -109,33 +113,50 @@ fn get_link_domain(link: &str) -> Result<String> {
 pub fn check_external_links(site: &Site) -> Result<()> {
     let library = site.library.write().expect("Get lock for check_external_links");
 
-    let mut all_links: Vec<(PathBuf, String, String)> = vec![];
+    struct LinkDef {
+        file_path: PathBuf,
+        external_link: String,
+        domain: String,
+    }
+
+    impl LinkDef {
+        pub fn new(file_path: PathBuf, external_link: String, domain: String) -> Self {
+            Self { file_path, external_link, domain }
+        }
+    }
+
+    let mut all_links: Vec<LinkDef> = vec![];
 
     for p in library.pages_values().into_iter() {
         for external_link in p.clone().external_links.into_iter() {
+            if should_skip_by_prefix(&external_link, &site.config.link_checker.skip_prefixes) {
+                continue;
+            }
+
             let domain = get_link_domain(&external_link)?;
-            all_links.push((p.file.path.clone(), external_link, domain));
+            all_links.push(LinkDef::new(p.file.path.clone(), external_link, domain));
         }
     }
 
     for s in library.sections_values().into_iter() {
         for external_link in s.clone().external_links.into_iter() {
+            if should_skip_by_prefix(&external_link, &site.config.link_checker.skip_prefixes) {
+                continue;
+            }
+
             let domain = get_link_domain(&external_link)?;
-            all_links.push((s.file.path.clone(), external_link, domain));
+            all_links.push(LinkDef::new(s.file.path.clone(), external_link, domain));
         }
     }
 
     println!("Checking {} external link(s).", all_links.len());
```

Comment:
Should we also print a count of skipped links?

Reply:
Yes

Reply:
Done https://github.com/getzola/zola/pull/1833/commits/b4fda105ac8ad83b5e7dd7a3a02c99f76664eb9a